packages/camera/camera/example/ios/RunnerTests/CameraPluginTests.m

```diff
@@ -0,0 +1,108 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import XCTest;
+
+#import <OCMock/OCMock.h>
+
+@interface CameraPlugin ()
+- (instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry
+                       messenger:(NSObject<FlutterBinaryMessenger> *)messenger;
+- (BOOL)handleMethodCallSync:(FlutterMethodCall *)call result:(FlutterResult)result;
+- (BOOL)handleMethodCallAsync:(FlutterMethodCall *)call result:(FlutterResult)result;
+@end
+
+@interface CameraPluginTests : XCTestCase
+@property(strong, nonatomic) id mockRegistrar;
+@property(strong, nonatomic) id mockMessenger;
+@property(strong, nonatomic) CameraPlugin *plugin;
+@end
+
+@implementation CameraPluginTests
+
+- (void)setUp {
+  [super setUp];
+  self.mockRegistrar = OCMProtocolMock(@protocol(FlutterPluginRegistrar));
+  self.mockMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  OCMStub([self.mockRegistrar messenger]).andReturn(self.mockMessenger);
+  self.plugin = [[CameraPlugin alloc] initWithRegistry:self.mockRegistrar
+                                             messenger:self.mockMessenger];
+}
+
+- (void)testHandleMethodCallSync_ShouldHandleSyncMethods {
+  id methodCallMock = OCMClassMock([FlutterMethodCall class]);
+  OCMStub([methodCallMock method]).andReturn(@"create");
```

Comment:
Why is this a mock? Mocking is a tool of last resort; `FlutterMethodCall` is trivial to instantiate.